### PR TITLE
fix: upgrade micromark-util-events-to-acorn

### LINKED ISSUE #380: [v2] incorrect comment loc info
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

2.0.0.next.1

### Link to runnable example

_No response_

### Steps to reproduce

````mdx
{/* First */}

<Story>
  {{
    template: /* HTML */ ``,
  }}
</Story>
````

print with `console.log(JSON.stringify(node.data?.estree.comments || [], null, 2))`

### Expected behavior

The second comment should start with `line: 5, column: 14` and end with `line: 5, column: 24`

### Actual behavior

```json
[
  {
    "type": "Block",
    "value": " First ",
    "start": 1,
    "end": 12,
    "loc": {
      "start": {
        "line": 1,
        "column": 1
      },
      "end": {
        "line": 1,
        "column": 12
      }
    },
    "range": [
      1,
      12
    ]
  }
]
[
  {
    "type": "Block",
    "value": " HTML ",
    "start": 42,
    "end": 52,
    "loc": {
      "start": {
        "line": 4,
        "column": 19
      },
      "end": {
        "line": 4,
        "column": 29
      }
    },
    "range": [
      42,
      52
    ]
  }
]
```

### Runtime

Node v16

### Package manager

yarn v1

### OS

macOS

### Build and bundle tools

_No response_

### PR DESCRIPTION
**Pull request by JounQin**
related https://github.com/micromark/micromark-extension-mdx-expression/pull/4

close #380

### CODE REVIEW

**Comment by JounQin** [comment]
@wooorm Any time to review?

File: packages/eslint-mdx/src/traverse.ts [hunk 1]
```diff
@@ -17,14 +17,14 @@ export class Traverse {
 
     const children = (node as Parent).children
 
+    this._enter(node, parent)
+
```
Comment (wooorm) [inline packages/eslint-mdx/src/traverse.ts:21]: Maybe your other code did depend on this. Maybe not.

If it did, you might have to make this configurable. But if it works, then your new code now matches NLR depth-first tree traversal, and that‘s used most often!  :+1: https://github.com/syntax-tree/unist#tree-traversal

**Review by wooorm — COMMENTED** [review]
This looks good to me! Is that all that’s needed to update to MDX 2? 🤔

File: packages/eslint-mdx/src/traverse.ts (hunk 1 above)
Comment (JounQin) [reply packages/eslint-mdx/src/traverse.ts:21]: The codes inside does not depend on the order except the comments we just knew. 🤣

**Comment by JounQin** [comment]
> This looks good to me! Is that all that’s needed to update to MDX 2? 🤔

https://github.com/mdx-js/eslint-mdx/issues/384

There is another known issue for now.

**Comment by JounQin** [comment]
@wooorm That issue doesn't block this PR, I think it's ready for a new `next` release?

**Comment by wooorm** [comment]
sure!

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
P.S. can you update deps (at least https://github.com/gajus/eslint-plugin-jsdoc/pull/880#issuecomment-1103673160) so everything builds on Node.js 18?

**Comment by wooorm** [comment]
Released 2.0.0-next.2!

**Pull request merged**